### PR TITLE
create manageState test-snap and interface

### PR DESCRIPTION
**Pull request by bowensanders**
This will add a test-snap for the manageState method.

### CODE REVIEW

File: packages/manageState/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        params: ['update', state],
+      });
+      return true;
+    case 'retrieveTestData':
+      //    return state;
+      const persistedData = await wallet.request({
+        method: 'snap_manageState',
+        params: ['get'],
+      });
+      return persistedData;
```
Comment (rekmarks) [inline packages/manageState/src/index.ts:39]: The linter forbids declaring variables in `case` blocks unless you wrap them like `case CASE: { /* content goes here */ }`. However, you don't even need to declare a variable in this case:
```suggestion
    case 'retrieveTestData':
      return await wallet.request({
        method: 'snap_manageState',
        params: ['get'],
      });
```

Comment (bowensanders) [reply packages/manageState/src/index.ts:39]: Oh sweet, thanks!

File: packages/manageState/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [0.1.3]
+### Changed
+- No Changes
```
Comment (rekmarks) [inline packages/manageState/CHANGELOG.md:11]: Changelogs are per-package, and should not contain changes for other packages in a monorepo. You should remove all versions except `0.1.3` from this changelog and update this to:
```suggestion
## [0.1.3]
### Added
- Initial release
```

File: packages/manageState/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+[0.1.2]: https://github.com/MetaMask/test-snaps/compare/v0.1.1...v0.1.2
+[0.1.1]: https://github.com/MetaMask/test-snaps/compare/v0.1.0...v0.1.1
+[0.1.0]: https://github.com/MetaMask/test-snaps/compare/v0.0.9...v0.1.0
+[0.0.9]: https://github.com/MetaMask/test-snaps/compare/v0.0.8...v0.0.9
+[0.0.8]: https://github.com/MetaMask/test-snaps/compare/v0.0.7...v0.0.8
+[0.0.7]: https://github.com/MetaMask/test-snaps/compare/v0.0.6...v0.0.7
+[0.0.6]: https://github.com/MetaMask/test-snaps/compare/v0.0.5...v0.0.6
+[0.0.5]: https://github.com/MetaMask/test-snaps/compare/v0.0.4...v0.0.5
+[0.0.4]: https://github.com/MetaMask/test-snaps/compare/v0.0.1...v0.0.4
+[0.0.1]: https://github.com/MetaMask/test-snaps/releases/tag/v0.0.1
```
Comment (rekmarks) [inline packages/manageState/CHANGELOG.md:82]:
```suggestion
[Unreleased]: https://github.com/MetaMask/test-snaps/compare/v0.1.3...HEAD
[0.1.3]: https://github.com/MetaMask/test-snaps/releases/tag/v0.1.3
```

**Review by rekmarks — DISMISSED** [review]

File: packages/manageState/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+- Added Error to index.html
+- Added instructions to README
+- Fixed build
+- Fixed linting
+- Fixed eslint ignorePattern
+- Added build to tsconfig exclude
+- Added icons
+- Changed outDir of typescript to `build`
+- The beginning of time
+
```
Comment (rekmarks) [inline packages/manageState/CHANGELOG.md:70]: These also need to go!

```suggestion
## [0.1.2]
### Changed
- No Changes

## [0.1.1]
### Changed
- No Changes

## [0.1.0]
### Added
- BIP44 snap for testing ([#20](https://github.com/MetaMask/test-snaps/pull/20))
- website build script ([#22](https://github.com/MetaMask/test-snaps/pull/22))
- various monorepo-related issues ([#18](https://github.com/MetaMask/test-snaps/pull/18))

### Changed
- Bump Node to 14, bump snaps-cli ([#21](https://github.com/MetaMask/test-snaps/pull/21))

## [0.0.9]
### Fixed
- workflow for release again

## [0.0.8]
### Fixed
- workflow for release

## [0.0.7]
### Added
- OpenRPC doc for confirm test snap ([#14](https://github.com/MetaMask/test-snaps/pull/14))

## [0.0.6]
### Fixed
- Added images icon to package.json ([#12](https://github.com/MetaMask/test-snaps/pull/12))

## [0.0.5]
### Fixed
- Fix publish files ([#10](https://github.com/MetaMask/test-snaps/pull/10))
- Make npm package public

## [0.0.4]
### Uncategorized
- Fixed linting

## [0.0.1]
### Uncategorized
- Changed version back to 0.0.0
- 0.0.1
- Fixed CHANGELOGs
- Changed confirm snap to be configurable via params
- Added Error to index.html
- Added instructions to README
- Fixed build
- Fixed linting
- Fixed eslint ignorePattern
- Added build to tsconfig exclude
- Added icons
- Changed outDir of typescript to `build`
- The beginning of time

```

**Review by bowensanders — COMMENTED** [review]
I believe all the changes have been made.

**Comment by bowensanders** [comment]
I'm pretty sure that all the requested changes have been made ?

File: packages/manageState/src/openrpc.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
+{
+  "openrpc": "1.2.4",
+  "info": {
+    "title": "Confirm Snap",
```
Comment (hmalik88) [inline packages/manageState/src/openrpc.json:4]: Why do we have this openrpc.json file here? This seems to be for the "Confirm" snap, was this copied over or auto generated?

Comment (bowensanders) [reply packages/manageState/src/openrpc.json:4]: oh my bad i thought i changed that. I did in fact use the confirm snap as a template
fixing now

**Comment by bowensanders** [comment]
I had taken openrpc out of the snap itself because it was unnecessary to function, if it needs to be added back in I can do that but it is gone as of right now

**Review by hmalik88 — APPROVED** [review]
LGTM!

**Pull request merged**